### ISSUE TITLE
Shortcut to mutate an element's children?

### ISSUE THREAD

**Issue by jturkel** [ohler55/ox#127]
Currently in OX it's a two step process to replace a node's children:

```
element.nodes.clear
element << new_child
```

This is a pretty common scenario (at least for my usage) and something that I miss coming from Nokogiri, how would you feel about about making this possible with a one-liner? 

The easiest way to support this would be to have `Ox::Element#nodes` lazy initialize the `@nodes` variable so we can write:

```
element.nodes.replace([new_child])
``` 

This also means we can use any of the `Array` mutation methods (e.g. `unshift`/`prepend`) without any special case handling of an element with no children

An `Ox::Element#text=` would also be convenient for scenarios where you want to replace all of a node's children with some text e.g.

```
my_element.text = 'hello world'
```

I'm happy to put together a PR with these changes if you think they're a good idea.

**Comment by ohler55** [comment]
I am not adverse to the idea. I'm not sure how a lazy init of @nodes will help though. Maybe I am missing something. I could see initializing @nodes to [] instead of nil though. I think thats what you are looking for.

The second feature of setting the nodes with a single text node is fine too although something more descriptive would be better. It should be clear that the old elements are being replaced. Maybe replace_text() or something.

If you submit a PR I'd be glad to approve once it looks right. I could also make the changes if you prefer. Either is ok. Lets agree first and then take either of those steps.

**Comment by jturkel** [comment]
Thanks for the fast response! The key for me is returning an array from `Element#nodes` that clients can always mutate to add/remove children (assuming you don't want to go down the road of forcing all mutation to happen via methods on `Element`). Either initializing `@nodes` to `[]` in the constructor or lazy initializing it in `Element#nodes` (i.e. `@nodes ||= []`) meets my needs. Lazy initialization would save a bit of memory for clients that don't access `nodes` but not sure that optimization is worthwhile. Can we also make a similar change to `@attributes` so we can get consistent mutation semantics for attributes and nodes?

`replace_text` sounds good to me.

That would be awesome if you want to make the changes. I can also take a look at it later in the week if you're busy.

**Comment by ohler55** [comment]
I'll make a pass at it tomorrow and ask you to review it before doing a release.

**Comment by jturkel** [comment]
Sounds great. Thanks again.

**Comment by ohler55** [comment]
I have a version on github with the requested changes. Please take a look when you get a chance.

**Comment by jturkel** [comment]
Thanks @ohler55! I added a few questions to the e4ca1c4cbbef306a14e4078547f5a06974f377f7 commit.

**Comment by ohler55** [comment]
Just released Ox 2.3.0.

**Comment by ohler55** [comment]
Can this be closed?
